src/ILCompiler.Compiler/src/Compiler/CompilerGeneratedMetadataManager.cs

```diff
@@ -127,6 +130,38 @@ public override bool IsReflectionBlocked(MetadataType type)
             }
         }
 
+        /// <summary>
+        /// Is there a reflection invoke stub for a method that is invokable?
+        /// </summary>
+        public override bool HasReflectionInvokeStubForInvokableMethod(MethodDesc method)
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// Gets a stub that can be used to reflection-invoke a method with a given signature.
+        /// </summary>
+        public override MethodDesc GetReflectionInvokeStub(MethodDesc method)
+        {
+            // Methods we see here shouldn't be canonicalized, or we'll end up creating bastardized instantiations
+            // (e.g. we instantiate over System.Object below.)
+            Debug.Assert(!method.IsCanonicalMethod(CanonicalFormKind.Any));
```

Comment:
We should probably kick this assert over to `InstantiateDynamicInvokeMethodForMethod`.